### PR TITLE
Fix dispatch time comparisons

### PR DESCRIPTION
**Pull request by lilyball**
This PR does two things:
1. It makes `<` comparisons on `DispatchWallTime` not crash.
2. It makes `<` comparisons against `DispatchTime.distantFuture` and `DispatchWallTime.distantFuture` behave as expected.

Fixes half of [SR-2807](https://bugs.swift.org/browse/SR-2807).

### CODE REVIEW

File: test/stdlib/Dispatch.swift [hunk 1]
```diff
@@ -85,3 +85,20 @@ DispatchAPI.test("dispatch_data_t deallocator") {
 		expectEqual(1, t)
 	}
 }
+
+DispatchAPI.test("DispatchTime comparisons") {
+    do {
+        let now = DispatchTime.now()
+        expectLT(now, .distantFuture)
```
Comment (kballard) [inline test/stdlib/Dispatch.swift]: Oops I didn't mean to leave this line in here. This was testing something else. I'm going to get rid of it.

File: stdlib/public/SDK/Dispatch/Time.swift [hunk 2]
```diff
@@ -73,8 +72,12 @@ public struct DispatchWallTime : Comparable {
 }
 
 public func <(a: DispatchWallTime, b: DispatchWallTime) -> Bool {
-	if a.rawValue == ~0 || b.rawValue == ~0 { return false }
-	return -Int64(a.rawValue) < -Int64(b.rawValue)
+    if b.rawValue == ~0 {
```
Comment (xwu) [inline stdlib/public/SDK/Dispatch/Time.swift]: Drive-by nit: indenting mismatch (this file uses \t).

**Comment by das** [comment]
@mwwa should review this.

The Linux overlay appears to have the same issue: https://github.com/apple/swift-corelibs-libdispatch/blob/master/src/swift/Time.swift

**Comment by kballard** [comment]
@das I already submitted swift-corelibs-libdispatch#182

File: stdlib/public/SDK/Dispatch/Time.swift (hunk 2 above)
Comment (kballard) [inline stdlib/public/SDK/Dispatch/Time.swift]: Oh weird, I have a Vim plugin that's supposed to auto-detect this sort of thing but apparently it doesn't work anymore >_<

Comment (kballard) [inline stdlib/public/SDK/Dispatch/Time.swift]: PR updated.

**Comment by kballard** [comment]
Is there anything blocking this from being reviewed?

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@kballard Thanks for bringing this up. I guess Matt just hasn't had time to review it yet.

**Comment by mwwa** [comment]
Yeah, I've been out of the country for two week. I'll try and get to this today/tomorrow.

**Comment by mwwa** [comment]
This looks good to me. @moiseev, can you merge?

**Comment by kballard** [comment]
As @das [suggested](https://github.com/apple/swift-corelibs-libdispatch/pull/182#issuecomment-251271019) on my other PR, this should probably be put into swift-3. Is there anything I need to do to make this happen?

**Comment by gparker42** [comment]
A PR with a cherrypick to the swift-3.0-branch may help push that forward.

**Comment by kballard** [comment]
Cherry-pick submitted as #5298.
